sftp/sftp.go

```diff
@@ -432,9 +434,50 @@ func (f *Fs) DirMove(src fs.Fs, srcRemote, dstRemote string) error {
 	return nil
 }
 
-// Hashes returns fs.HashNone to indicate remote hashing is unavailable
 func (f *Fs) Hashes() fs.HashSet {
-	return fs.HashSet(fs.HashNone)
+	if f.cachedHashes != nil {
+		return *f.cachedHashes
+	}
+
+	session, err := f.sshClient.NewSession()
+	if err != nil {
+		return fs.HashSet(fs.HashNone)
+	}
+	sha1Output, err := session.Output("echo 'abc' | sha1sum | awk '{ print $1 }'")
```

Comment:
I've dropped the dependency on `awk`, but I think performing a sanity check with `md5sum`/`sha1sum` is the safer option - nobody knows if the binary actually does what we're expecting. I also can't think of any system running rclone without `echo` support, so this should be fine.